pwndbg/dbg/lldb/repl/__init__.py

```diff
@@ -679,60 +687,96 @@ def process_launch(driver: ProcessDriver, relay: EventRelay, args: List[str], db
     if driver.has_process():
         print(message.error("error: a process is already being debugged"))
         return
 
-    # The first two arguments - executable name and wait_for_launch - don't
-    # matter, we set them later. The third one is required, as it tells LLDB the
-    # attach should be asynchronous.
-    params = lldb.SBAttachInfo(None, False, True)
-
-    if args.name is not None:
-        params.SetExecutable(args.name)
-    if args.pid is not None:
-        params.SetProcessID(args.pid)
-    params.SetWaitForLaunch(args.waitfor)
-    if getattr(args, "continue"):
-        params.SetResumeCount(1)
-    params.SetIgnoreExisting(not args.include_existing)
-
     io_driver = get_io_driver()
+
     result = driver.attach(
         dbg.debugger.GetTargetAtIndex(0),
         io_driver,
-        params,
+        info,
     )
 
     if not result.success:
         print(message.error(f"Could not attach to process: {result.description}"))
         return
 
     # Continue execution if the user has requested it.
-    if getattr(args, "continue"):
+    if cont:
         # Same logic applies here as in `process_launch`.
         relay._set_ignore_resumed(1)
-
         driver.cont()
     else:
         # Same logic applies here as in `process_launch`.
         dbg._trigger_event(EventType.STOP)
 
 
+def process_attach(driver: ProcessDriver, relay: EventRelay, args: List[str], dbg: LLDB) -> None:
+    """
+    Attaches to a process with the given arguments.
+    """
+    args = parse(args, process_attach_ap, process_attach_unsupported)
+    if not args:
+        return
+
+    # The first two arguments - executable name and wait_for_launch - don't
+    # matter, we set them later. The third one is required, as it tells LLDB the
+    # attach should be asynchronous.
+    info = lldb.SBAttachInfo(None, False, True)
+
+    if args.name is not None:
+        info.SetExecutable(args.name)
+    if args.pid is not None:
+        info.SetProcessID(args.pid)
+    info.SetWaitForLaunch(args.waitfor)
+
+    do_continue = getattr(args, "continue", False)
+    if do_continue:
+        info.SetResumeCount(1)
+    info.SetIgnoreExisting(not args.include_existing)
+
+    _attach_with_info(driver, relay, dbg, info, cont=do_continue)
+
+
+def attach(driver: ProcessDriver, relay: EventRelay, args: List[str], dbg: LLDB) -> None:
+    """
+    Attaches to a process with the given name or pid based on regex match.
+    Used for `_regexp-attach <pid|name>` (alias for `attach <pid|name>`)
+    Note: for some reason, `attach` does not really take a regex for process name.
+    """
+    assert len(args) == 1, "Expected 1 argument: <pid> or <name>"
+    arg = args[0]
+
+    # exec name - None, we set it later (or pid)
+    # wait_for_launch - False, since we don't wait
+    # third arg - tell LLDB to attach asynchronously
+    info = lldb.SBAttachInfo(None, False, True)
+
+    # Argument is pid
+    if arg.isdigit():
```

Comment:
This check is nice bcoz `try: int(...)` would allow some incorrect values:
```
In [1]: '-1'.isdigit()
Out[1]: False

In [2]: '11'.isdigit()
Out[2]: True

In [3]: '+11'.isdigit()
Out[3]: False

In [4]: '1.1'.isdigit()
Out[4]: False
```

This one here only allows for `000` which is invalid pid, but whatever:
```
In [5]: '000'.isdigit()
Out[5]: True
```